test/setup.js

```diff
@@ -12,36 +12,80 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import should from 'should';
 import sinon from 'sinon';
-import 'should-sinon';
+import chai, { Assertion } from 'chai';
+import 'chai/register-should';
+import chaiAsPromised from 'chai-as-promised';
+import sinonChai from 'sinon-chai';
 import naclFactory from 'js-nacl';
 
 process.env.NODE_ENV = 'test';
 
-should.use((_, Assertion) => {
-	Assertion.add('hexString', function hexString() {
-		this.params = {
-			operator: 'to be hex string',
-		};
-		Buffer.from(this.obj, 'hex')
-			.toString('hex')
-			.should.equal(this.obj);
-	});
+Assertion.addMethod('hexString', function handleAssert(type) {
+	// eslint-disable-next-line no-underscore-dangle
+	const obj = this._obj;
 
-	Assertion.add('integer', function integer() {
-		this.params = {
-			operator: 'to be an integer',
-		};
-		parseInt(this.obj, 10).should.equal(this.obj);
-	});
+	// eslint-disable-next-line no-underscore-dangle
+	new Assertion(this._obj).to.be.a('string');
+
+	// second, our type check
+	this.assert(
+		// eslint-disable-next-line no-underscore-dangle
+		obj._type === type,
+		'expected #{this} to be of type #{exp} but got #{act}',
+		'expected #{this} to not be of type #{act}',
+		type,
+		// eslint-disable-next-line no-underscore-dangle
+		obj._type,
+	);
+	// eslint-disable-next-line no-underscore-dangle
+	const converted = Buffer.from(this._obj, 'hex').toString('hex');
+	this.assert(
+		converted === obj,
+		'expected #{this} to be of hexString #{exp} but got #{act}',
+		'expected #{this} to not be of type #{act}',
+		obj,
+		converted,
+	);
+});
+
+Assertion.addMethod('integer', function handleAssert(type) {
+	// eslint-disable-next-line no-underscore-dangle
+	const obj = this._obj;
+
+	// first, our instanceof check, shortcut
+	// eslint-disable-next-line no-underscore-dangle
+	new Assertion(this._obj).to.be.a('number');
+
+	// second, our type check
+	this.assert(
+		// eslint-disable-next-line no-underscore-dangle
+		obj._type === type,
+		'expected #{this} to be of type #{exp} but got #{act}',
+		'expected #{this} to not be of type #{act}',
+		type,
+		// eslint-disable-next-line no-underscore-dangle
+		obj._type,
+	);
+	// eslint-disable-next-line no-underscore-dangle
+	const converted = parseInt(this._obj, 10);
+	this.assert(
+		converted === obj,
+		'expected #{this} to be of hexString #{exp} but got #{act}',
+		'expected #{this} to not be of type #{act}',
+		obj,
+		converted,
+	);
```

Comment:
Can this be simplified?

Reply:
Mostly following 
[http://chaijs.com/guide/helpers/#adding-methods](http://chaijs.com/guide/helpers/#adding-methods)
I need to dig deeper to see if this can be simplified.

Reply:
Second thought, this shouldn't be method, I will change this to property.
like default
`to.be.true()` => `to.be.true`